bundles/org.openhab.binding.sunsynk/src/main/java/org/openhab/binding/sunsynk/internal/handler/SunSynkAccountHandler.java

```diff
@@ -0,0 +1,134 @@
+    private @Nullable ScheduledFuture<?> discoverApiKeyJob;
+
+    public SunSynkAccountHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        logger.debug("SunSynk Handler Intialised attempting to retrieve configuration");
+        startDiscoverApiKeyJob();
+    }
+
+    private void startDiscoverApiKeyJob() {
+        if (discoverApiKeyJob == null || discoverApiKeyJob.isCancelled()) {
+            Runnable runnable = new Runnable() {
+                @Override
+                public void run() {
+                    logger.debug("Starting account discovery job.");
+                    configAccount();
+                    logger.debug("Done account discovery job.");
+                }
+            };
+            discoverApiKeyJob = scheduler.schedule(runnable, 1, TimeUnit.SECONDS);
+        }
+    }
+
+    public void setBridgeOnline() {
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    public void setBridgeOffline() {
+        updateStatus(ThingStatus.OFFLINE);
+    }
+
+    public List<Inverter> getInvertersFromSunSynk() {
+        logger.debug("Attempting to find inverters tied to account");
+        ArrayList<Inverter> inverters = new ArrayList<>();
+        try {
+            inverters = sunAccount.getDetails();
+        } catch (SunSynkInverterDiscoveryException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Error attempting to find inverters registered to account");
+        }
+        return inverters;
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing sunsynk bridge handler.");
+        if (discoverApiKeyJob != null && !discoverApiKeyJob.isCancelled()) {
+            discoverApiKeyJob.cancel(true);
+            discoverApiKeyJob = null;
+        }
+    }
+
+    public void configAccount() {
+        SunSynkAccountConfig accountConfig = getConfigAs(SunSynkAccountConfig.class);
+        try {
+            this.sunAccount.authenticate(accountConfig.getEmail(), accountConfig.getPassword());
+        } catch (SunSynkAuthenticateException | SunSynkTokenException e) {
+            logger.debug("Error attempting to autheticate account Msg = {} Cause = {}", e.getMessage().toString(),
+                    e.getCause().toString());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Error attempting to authenticate account");
+            return;
+        }
+        updateStatus(ThingStatus.ONLINE);
+        logger.debug("Account configuration updated : {}", this.sunAccount.toString());
+    }
+
+    public boolean refreshAccount() throws SunSynkAuthenticateException {
+        SunSynkAccountConfig accountConfig = getConfigAs(SunSynkAccountConfig.class);
```

Comment:
Cache the SunSynkAccountConfig accountConfig at class level when initilizing this class.

Reply:
Done